### PR TITLE
add_rpc_calls_and_fix_tests

### PR DESCRIPTION
**Pull request by reginaldo-binary**

### CODE REVIEW

File: lib/Net/Async/Blockchain/Client/RPC/BTC.pm [hunk 1]
```diff
@@ -137,5 +137,56 @@ sub get_block_hash {
     return $self->_request('getblockhash', @params);
 }
 
+=head2 get_balance
+
+https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/blockchain/getbalance/
```
Comment (konstantinos-deriv) [inline lib/Net/Async/Blockchain/Client/RPC/BTC.pm:142]: page not found for this comment

File: lib/Net/Async/Blockchain/Client/RPC/BTC.pm [hunk 2]
```diff
@@ -137,5 +137,56 @@ sub get_block_hash {
...
+=cut
+
+sub get_balance {
+    my ($self, @params) = @_;
+    return $self->_request('getbalance', @params);
+}
+
+=head2 get_balances
+
+https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/blockchain/getbalances/
```
Comment (konstantinos-deriv) [inline lib/Net/Async/Blockchain/Client/RPC/BTC.pm:159]: ditto

File: lib/Net/Async/Blockchain/Client/RPC/BTC.pm [hunk 3]
```diff
@@ -137,5 +137,56 @@ sub get_block_hash {
...
+=cut
+
+sub get_balances {
+    my ($self, @params) = @_;
+    return $self->_request('getbalances', @params);
+}
+
+=head2 list_received_by_address
+
+https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/blockchain/listreceivedbyaddress/
```
Comment (konstantinos-deriv) [inline lib/Net/Async/Blockchain/Client/RPC/BTC.pm:176]: ditto

File: lib/Net/Async/Blockchain/Omni.pm [hunk 4]
```diff
@@ -146,13 +146,18 @@ async sub _process_transaction {
...
     my ($to_response, $from_response);
 
-    $from_response = await $self->rpc_client->list_by_addresses($from->{address});
+    # parameters as the following sequence for listing the received by address
+    # minimum confirmation 1
+    # include empty false
+    # include watch only  true
+
+    $from_response = await $self->rpc_client->list_received_by_address(1, \0, \1, $from->{address});
     if ($from_response && @$from_response) {
```
Comment (konstantinos-deriv) [inline lib/Net/Async/Blockchain/Omni.pm:155]: super minor but you can write this as `$from_response->@*`

File: t/memory.t [hunk 5]
```diff
@@ -24,27 +24,31 @@ my $accounts =
 
 my %accounts = map { lc($_) => 1 } $accounts->@*;
 
+my $decoded_transaction = decode_json_utf8(
```
Comment (konstantinos-deriv) [inline t/memory.t:27]: just as a general comment, would be nice if we had a specific file for the data that we use so we wouldn't have the need to hardcode json structures like that in the test files.

File: lib/Net/Async/Blockchain/Omni.pm (hunk 4 above)
Comment (reginaldo-binary) [reply lib/Net/Async/Blockchain/Omni.pm:155]: here we expect that the $from_response can maybe be `undef` so we need to check the variable before converting it to an array

**Pull request closed without merging**